src/parallelism/threads/exec.js

```diff
@@ -10,15 +10,16 @@ const executeLifecycle = {
     },
     test: async () => {
         // eslint-disable-next-line no-constant-condition
-        while (true) {
+        while (!process.env.WARTHOG_END) {
             // eslint-disable-next-line no-loop-func
             Array.from({ length: workerData.iterations }).forEach(async () => {
                 try {
                     const duration = await getPerformance(workerData.testLifecycle.test);
                     parentPort.postMessage({ iteration: { duration } });
                 } catch (error) {
-                    console.log(error);
+                    console.error(error);
                     parentPort.postMessage({ iteration: {} });
+                    process.env.WARTHOG_END = true;
```

Comment:
Is it better to write to a system env variable than creating a local one?

Reply:
The value of this var is going to be changing dynamically, by default it's not defined.
Since this condition happens, when an `undefined` is obtained trying to read that bar the process execution will continue.
But when this variable becomes declared with the value `true`, the next [loop](https://github.com/onebeyond/warthog-load-testing/pull/2/files#diff-59c9eaa5e2f0a440a6ddb91416d76cdb75b3f881a0c838600f13e7db7e881280R13) execution will not be performed.

Since we need to change the value of this var from the [script test](https://github.com/onebeyond/warthog-load-testing/pull/2/files#diff-c3fbdf9d6ee2400d5559cd626823d18863fffaeb63656d27b9fa996595f217fb) being executed inside that function that contains the loop mentioned before. The test can not access local vars, it only shares the process variables. Also that gives us to control that loop execution from the testing library in a simple way without any message queue system.

Reply:
Thanks for the explanation!